packages/sqs/test/publishers/SqsPermissionPublisher.spec.ts

```diff
@@ -196,20 +224,32 @@ describe('SqsPermissionPublisher', () => {
 
       await permissionPublisher.publish(message)
 
-      const spy = await permissionPublisher.handlerSpy.waitForMessageWithId('1', 'published')
-      expect(spy.message).toEqual({ ...message, timestamp: expect.any(String) })
-      expect(spy.processingResult).toBe('published')
+      await waitAndRetry(() => !!receivedMessage)
```

Comment:
can't we use `await myConsumer.handlerSpy.waitForMessage({
    // internal fields described here
})`
instead?

Reply:
not sure if they are exposed to the handlerSpy, though

Reply:
No, the number of retries is an internal property and it shouldn't be returned on the spy as it can break tests if someone is using `toEqual` (as we were doing on library tests)

Reply:
gotcha! all good then